### PR TITLE
Add phone number field on registration form, controlled by a setting

### PR DESCRIPTION
**Pull request by ejucovy**

### CODE REVIEW

**Comment by tobiasmcnulty** [comment]
Any chance you could add a couple tests for this in `test_registration.py`? E.g., one that forces the setting to `True` and makes sure the phone number gets saved and another that forces it to `False` and makes sure the field doesn't show up? While you're at it, might be worthwhile to do the same for `display_name` -- I was surprised when that PR didn't require any test changes (though I neglected to mention it at the time!).

File: opendebates/forms.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from django.utils.html import mark_safe
 from django.utils.translation import ugettext_lazy as _
 from localflavor.us.forms import USZipCodeField
+from localflavor.us.forms import USPhoneNumberField
```
Comment (tobiasmcnulty) [inline opendebates/forms.py]: Could combine this with the line above.

File: opendebates/models.py [hunk 2]
```diff
@@ -271,6 +272,7 @@ def user_display_name(self):
 
     display_name = models.CharField(max_length=255, null=True, blank=True)
     twitter_handle = models.CharField(max_length=255, null=True, blank=True)
+    phone_number = PhoneNumberField(null=True, blank=True)
```
Comment (tobiasmcnulty) [inline opendebates/models.py]: We seem to repeat this pattern elsewhere, but anything that becomes a `varchar` in the DB shouldn't have `null=True` (form behavior should remain the same, but you will have to provide a default value of `''` when creating the migration).

File: opendebates/static/js/registration.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
 (function() {
   $(window).load(function() {
     $(".form-group").each(function() {
-      if ($(this).find("label").length === 1) {
-        var text = $(this).find("label").hide().text();
+      var $label = $(this).find("label");
+      if ($label.length === 1) {
+        var text = $label.text();
```
Comment (tobiasmcnulty) [inline opendebates/static/js/registration.js:6]: I'm not sure I understand the reason for the changes here -- can you help me understand what's going on? `hide()` was lost on this line but I'm not sure if that was intentional or not.

Comment (tobiasmcnulty) [reply opendebates/static/js/registration.js:6]: AFAICT tool tips were broken and this PR brings them back -- is that right?

Comment (ejucovy) [reply opendebates/static/js/registration.js:6]: Yup, that's right.  Tooltips were previously completely hidden, because we didn't want field labels, just placeholders.  But we need a tooltip to explain why we're asking for a phone number so that we don't freak people out.  So this PR continues to take any text content from the label and put it in a placeholder instead, but retains the tooltip, and positions the tooltip "inside" the form field with some css positioning. 

Basically it turns

```
<label>Foo! Bar! Baz! <span>Morx</span></label>
<input>
```

into

```
<label><span>Morx</span></label>
<input placeholder="Foo! Bar! Baz!">
```

**Comment by ejucovy** [comment]
Yeah - I had skipped writing tests because it was looking like it would be harder and uglier than I expected.  I just pushed a commit with tests though.  The problem is that (a) the determination of what form fields exist happens at import time, not runtime, and (b) the view stores a reference to the form class at import time too.  So I ended up force-reloading the form class and also manually updating the reference on the view class.  I can't think of a cleaner way to do it. :-/

**Comment by tobiasmcnulty** [comment]
Yikes! Right. The cleaner way is probably to put the field logic in `__init__()`, however, if this works (it appears to!) & you're out of time to work on this now we can refactor later.

**Comment by tobiasmcnulty** [comment]
Just wanted to make sure you're not working on this. I'll give it a final review & merge if not!

**Comment by ejucovy** [comment]
Yeah - let's do that cleanup later. (Sorry! busy day!!)

**Comment by tobiasmcnulty** [comment]
No prob!

🚀
